docs/guide/output-data-widgets.md

```diff
@@ -37,6 +37,90 @@ echo DetailView::widget([
 ]);
 ```
 
+Remember that unlike [[yii\widgets\GridView|GridView]] which processes a set of models,
+[[yii\widgets\DetailView|DetailView]] processes just one. So most of the times there is no need for using closure since
+`$model` is the only one model for display and available in view as variable.
+
+However some cases can make using of closure useful, for example:
+
+```php
+echo DetailView::widget([
+    'model' => $model,
+    'attributes' => [
+        [
+            'attribute' => 'owner',
+            'value' => $model->owner->name,
+            'visible' => \Yii::$app->user->can('posts.owner.view'),
+        ],
+    ],
+]);
+```
+
+In this example regardless of `visible` value, including case when user does not have permission to view owner
+(`visible` evaluates to `false`), the `value` will be always calculated by accessing relation and its property
+(`$model->owner->name`). Here it can cause additional SQL query to execute because lazy loading is used by default.
+The `value` can contain other calculations which are undesired and useless if as a result this attribute won't be
+displayed at all. To prevent that, use closure:
+
+```php
+echo DetailView::widget([
+    'model' => $model,
+    'attributes' => [
+        [
+            'attribute' => 'owner',
+            'value' => function ($model) {
+                return $model->owner->name;
+            },
+            'visible' => \Yii::$app->user->can('posts.owner.view'),
+        ],
+    ],
+]);
+```
+
+Closure can be also useful in case when calculation logic of displayed value can't be written in one single line:
+
+```php
+echo DetailView::widget([
+    'model' => $model,
+    'attributes' => [
+        [
+            'attribute' => 'owner',
+            'value' => function ($model) {
+                if ( ... ) {
+                    // Some calculations
+                    return ...;
+                } elseif ( ... ) {
+                    // Some calculations
+                    return ...;
+                } else {
+                    // Some calculations
+                    return ...;
+                }
+            },
+        ],
+    ],
+]);
+```
+
+Without closure it would be less readable or additional variables would be required. However remember that in case of
+complex presentation logic sometimes it's better to use decorators or at least additional helpers. Putting it in models
+is not always a good idea because it's more related with display (view layer). And additional benefit of that is when
+you have common attributes and values in both `GridView` and `DetailView` you can avoid copy paste:
+
+```php
+echo DetailView::widget([
+    'model' => $model,
+    'attributes' => [
+        [
+            'attribute' => 'owner',
+            'value' => // Call helper or decorator here instead
+        ],
+    ],
+]);
+```
+
```

Comment:
Not sure about adding reference to https://github.com/drapergem/draper. Can't find something similar for PHP. But at least brief mention of it is useful in my opinion, because I found people often put complex logic there and copy paste it from `GridView`.